src/views/VaultDetails.tsx

```diff
@@ -0,0 +1,251 @@
+
+import { getNumberSign } from '@/lib/numbers';
+
+export const VaultDetails: React.FC = () => {
+  const stringGetter = useStringGetter();
+  const { market, configs } = useAppSelector(getCurrentMarketData, shallowEqual) ?? {};
+  const { id, name } = useAppSelector(getCurrentMarketAssetData, shallowEqual) ?? {};
+  const { allTimePnl, currentLeverageMultiple, thirtyDayReturnPercent, currentPosition } =
+    useAppSelector(getCurrentMarketVaultDetails) ?? {};
+  const { totalValue } = useAppSelector(getCurrentMarketVaultMetadata) ?? {};
+  const { vaultsLearnMore } = useURLConfigs();
+
+  const items = [
+    {
+      key: 'balance',
+      label: stringGetter({ key: STRING_KEYS.VAULT_BALANCE }),
+      value: <Output value={totalValue} type={OutputType.CompactFiat} />,
+    },
+    {
+      key: 'all-time-pnl',
+      label: stringGetter({ key: STRING_KEYS.VAULT_ALL_TIME_PNL }),
+      value: (
+        <$ColoredReturn $sign={getNumberSign(allTimePnl?.absolute)}>
+          {allTimePnl?.absolute != null ? (
+            <Output value={allTimePnl?.absolute} type={OutputType.CompactFiat} />
+          ) : (
+            '-'
+          )}{' '}
+          <Output value={allTimePnl?.percent} type={OutputType.Percent} withParentheses />
+        </$ColoredReturn>
+      ),
+    },
+    {
+      key: '30d-apr',
+      label: stringGetter({ key: STRING_KEYS.VAULT_THIRTY_DAY_APR }),
+      value: <Output value={thirtyDayReturnPercent} type={OutputType.Percent} />,
+    },
+    {
+      key: 'vault-position',
+      label: stringGetter({ key: STRING_KEYS.VAULT_POSITION }),
+      value: (
+        <$PositionContainer>
+          <$ColoredReturn $sign={getNumberSign(currentPosition?.asset)}>
+            <Output
+              value={currentPosition?.asset}
+              type={OutputType.Asset}
+              tag={id}
+              showSign={ShowSign.Negative}
+              fractionDigits={configs?.stepSizeDecimals ?? TOKEN_DECIMALS}
+            />
+          </$ColoredReturn>
+          <$MutedText>
+            <Output
+              value={currentPosition?.usdc}
+              type={OutputType.Fiat}
+              withParentheses
+              fractionDigits={USD_DECIMALS}
+            />
+          </$MutedText>
+        </$PositionContainer>
+      ),
+    },
+    {
+      key: 'vault-leverage',
+      label: stringGetter({ key: STRING_KEYS.VAULT_LEVERAGE }),
+      value: <Output value={currentLeverageMultiple} type={OutputType.Multiple} />,
+    },
+  ];
+
+  return (
+    <$VaultDetails>
+      <$Header>
+        <$WrapRow>
+          <$MarketTitle>
+            <AssetIcon symbol={id} />
+            {stringGetter({ key: STRING_KEYS.ASSET_VAULT, params: { ASSET: name } })}
+          </$MarketTitle>
+        </$WrapRow>
+
+        <$VaultDescription>
+          <p>{stringGetter({ key: STRING_KEYS.VAULT_DESCRIPTION })}</p>
+        </$VaultDescription>
+
+        <$Buttons>
+          <Link to={`${AppRoute.Trade}/${market}`}>
```

Comment:
Should we add a `ButtonType.InternalLink` that incorporates the Link component?

Reply:
I don't think it's super worth it since we use them so little and it has a completely different API from normal links so we'd have to add like 3 more props to all the Button interfaces.